beacon-chain/blockchain/process_attestation.go

```diff
@@ -20,7 +20,8 @@ import (
 var ErrTargetRootNotInDB = errors.New("target root does not exist in db")
 
 // onAttestation is called whenever an attestation is received, verifies the attestation is valid and saves
-/// it to the DB.
+// it to the DB. As a pure function, this does not delay attestation based on the spec outline. The delay is handled
```

Comment:
What does "pure function" mean in this context?

Reply:
Stateless, it doesn't "hold" attestations within the function. I'll update the comment to make it more clear

Reply:
OK thanks. To me, a pure function has no dependencies
```go
func pureFn(a, b int) int {
  return a + b
}

func notPureFn(a, b int) int {
  return db.Lookup(a, b)
}
```